From a73624d846242c4cf6e90ab8802efb22ec3408b0 Mon Sep 17 00:00:00 2001
From: Simone Schwizer <simone.schwizer@bytesatwork.ch>
Date: Wed, 3 Sep 2025 14:21:14 +0200
Subject: [PATCH] bytesatwork: bawconfig: Update bawconfig

Used cherry picked commits:
23bc62ffe9b bytesatwork: bawconfig: Add stm32mp2 HW revision and ram type
de8bc8ca311 bytesatwork: bawconfig: Remove HW dependencies

Signed-off-by: Simone Schwizer <simone.schwizer@bytesatwork.ch>
---
 board/bytesatwork/baw_config/baw_config.c     |  3 +
 board/bytesatwork/baw_config/baw_config.h     |  3 +
 board/bytesatwork/baw_config/baw_config_get.c | 63 +------------------
 3 files changed, 7 insertions(+), 62 deletions(-)

diff --git a/board/bytesatwork/baw_config/baw_config.c b/board/bytesatwork/baw_config/baw_config.c
index 2a63fe3810..f223e488a6 100644
--- a/board/bytesatwork/baw_config/baw_config.c
+++ b/board/bytesatwork/baw_config/baw_config.c
@@ -24,6 +24,8 @@ const struct baw_config_name_pair baw_config_pcb_name[] = {
 	{ M9_PCB_REV_1_0, "M9 Rev 1.0" },
 	{ M9_PCB_REV_2_0, "M9 Rev 2.0" },
 	{ M7_PCB_REV_1_1, "M7 Rev 1.1" },
+	{ M10_PCB_REV_0_1, "M10 Rev 0.1" },
+	{ M10_PCB_REV_1_0, "M10 Rev 1.0" },
 	{ -1, NULL },
 };
 
@@ -58,6 +60,7 @@ const struct baw_config_name_pair baw_config_ram_name[] = {
 	{ M9_RAM_MT53E1G16D1FW_046, "MT53E1G16D1FW-046 2 GB LPDDR4X @ 1866 MHz" },
 	{ M7_RAM_MT53E1536M32D4DE_046, "MT53E1536M32D4DE-046 6 GB LPDDR4 @ 2000 MHz"},
 	{ M7_RAM_IS43LQ32512A_046, "IS43LQ32512A-046BLI 2GB LPDDR4 @ 2000 MHz"},
+	{ M10_RAM_MT53D512M16D1DS_046, "MT53D512M16D1DS-046 1 GB LPDDR4 @ 1200 MHz"},
 	{ -1, NULL },
 };
 
diff --git a/board/bytesatwork/baw_config/baw_config.h b/board/bytesatwork/baw_config/baw_config.h
index 8a86b4a9c2..4c741db9f3 100644
--- a/board/bytesatwork/baw_config/baw_config.h
+++ b/board/bytesatwork/baw_config/baw_config.h
@@ -28,6 +28,8 @@ enum baw_config_pcb {
 	M9_PCB_REV_1_0 = 15,
 	M9_PCB_REV_2_0 = 16,
 	M7_PCB_REV_1_1 = 17,
+	M10_PCB_REV_0_1 = 18,
+	M10_PCB_REV_1_0 = 19,
 };
 
 enum baw_config_ram {
@@ -64,6 +66,7 @@ enum baw_config_ram {
 	M9_RAM_MT53E1G16D1FW_046 = 28,
 	M7_RAM_MT53E1536M32D4DE_046 = 29,
 	M7_RAM_IS43LQ32512A_046 = 30,
+	M10_RAM_MT53D512M16D1DS_046 = 31,
 };
 
 enum baw_config_flash {
diff --git a/board/bytesatwork/baw_config/baw_config_get.c b/board/bytesatwork/baw_config/baw_config_get.c
index effc6ee785..2a8091947a 100644
--- a/board/bytesatwork/baw_config/baw_config_get.c
+++ b/board/bytesatwork/baw_config/baw_config_get.c
@@ -27,51 +27,6 @@ void baw_config_get(struct baw_config *config)
 		return;
 	}
 
-	if (IS_ENABLED(CONFIG_SPL_BUILD) && IS_ENABLED(CONFIG_TARGET_BYTEENGINE_AM335X) &&
-	    IS_ENABLED(CONFIG_DM_I2C)) {
-		struct udevice *pmic;
-		struct udevice *bus;
-		int ret;
-
-		ret = uclass_get_device_by_seq(UCLASS_I2C, CONFIG_BAW_CONFIG_EEPROM_BUS, &bus);
-		if (ret) {
-			printf("%s(): uclass_get_device_by_seq(): %d\n", __func__, ret);
-			goto default_config;
-		}
-
-		ret = dm_i2c_probe(bus, PMIC_ADDRESS, 0, &pmic);
-		if (ret) {
-			printf("%s(): dm_i2c_probe(): %d\n", __func__, ret);
-			goto default_config;
-		}
-
-		ret = i2c_set_chip_offset_len(pmic, 1);
-		if (ret) {
-			printf("%s(): i2c_set_chip_offset_len(): %d\n", __func__, ret);
-			goto default_config;
-		}
-
-		ret = dm_i2c_read(pmic, 0x20, &reg, 1);
-		if (ret) {
-			printf("Error: PMIC read failed\n");
-			goto default_config;
-		} else {
-			if (reg == 0x05) {
-				config->ram = M2_RAM_MT47H128M16RT25E;	/* set DDR2 */
-				printf("DDR2 detected\n");
-			} else if (reg == 0x01) {
-				config->ram = M2_RAM_K4B2G1646EBIH9;	/* set legacy DDR3 */
-				printf("DDR3 detected\n");
-			} else {
-				goto default_config;
-			}
-		}
-
-		return;
-	}
-
-default_config:
-
 	if (IS_ENABLED(CONFIG_BAW_CONFIG_BUILTIN)) {
 		config->ram = BAW_CONFIG_BUILTIN_RAM;
 		if (IS_ENABLED(CONFIG_SPL_BUILD)) {
@@ -80,25 +35,9 @@ default_config:
 
 			return;
 		}
-	} else {
-		if (IS_ENABLED(CONFIG_TARGET_BYTEENGINE_AM335X))
-			config->ram = M2_RAM_K4B2G1646EBIH9;	/* set default to legacy DDR3 */
-		else if (IS_ENABLED(CONFIG_TARGET_STM32MP1_BYTEDEVKIT))
-			config->ram = M5_RAM_K4B4G1646DBIK0;
-		else if (IS_ENABLED(CONFIG_TARGET_IMX8MM_BYTEDEVKIT))
-			config->ram = M6_RAM_MT53E128M32D2DS_053;
-		else if (IS_ENABLED(CONFIG_TARGET_AM62X_BYTEDEVKIT_A53))
-			config->ram = M8_RAM_MT53E256M16D1DS_046;
-		else if (IS_ENABLED(CONFIG_TARGET_AM62PX_BYTEDEVKIT_A53))
-			config->ram = M8P_RAM_MT53E2G32D4DE_046;
-		else if (IS_ENABLED(CONFIG_TARGET_IMX8MP_BYTEDEVKIT))
-			config->ram = M7_RAM_MT53E1G32D2FW_046;
-		else if (IS_ENABLED(CONFIG_TARGET_IMX93_BYTEDEVKIT))
-			config->ram = M9_RAM_MT53D512M16D1DS_046;
 	}
 
 	if (IS_ENABLED(CONFIG_SPL_BUILD)) {
-		printf("Error: no RAM configuration found, trying fallback config ");
-		printf("%u (%s)\n", config->ram, baw_config_get_ram_name(config->ram));
+		printf("Error: no RAM configuration found");
 	}
 }
